SampleScripts/ppm/passwordpolicymanager.ps1

```diff
@@ -0,0 +1,1456 @@
+
+		#parsing ESXi password policy string
+		$result.PasswordQualityControl | Select-String -Pattern "^retry=(\d+)\s+min=(.+),(.+),(.+),(.+),(.+)" | Foreach-Object {$PasswdPolicyRetryValue, $PasswdPolicyMinValue1, $PasswdPolicyMinValue2, $PasswdPolicyMinValue3, $PasswdPolicyMinValue4, $PasswdPolicyMinValue5 = $_.Matches[0].Groups[1..6].Value}
+		$esxiConfigTypeValues.passwdExpInDays = $result.PaswordMaxDays
+		$esxiConfigTypeValues.passwdMaxFailAttempts = $PasswdPolicyRetryValue
+		$esxiConfigTypeValues.passwdMinimumLengthFor1CharClass = $PasswdPolicyMinValue1
+		$esxiConfigTypeValues.passwdMinimumLengthFor2CharClass = $PasswdPolicyMinValue2
+		$esxiConfigTypeValues.passwdMinimumLengthFor3CharClass = $PasswdPolicyMinValue4
+		$esxiConfigTypeValues.passwdMinimumLengthFor4CharClass = $PasswdPolicyMinValue5
+		$esxiConfigTypeValues.passwdMinimumCharLengthForPhrase = $PasswdPolicyMinValue3
+
+		$nodeConfigValue = New-Object -TypeName psobject
+		$nodeConfigValue | Add-Member -notepropertyname "FQDN" -notepropertyvalue $result.fqdn
+		$nodeConfigValue | Add-Member -notepropertyname "productType" -notepropertyvalue "ESXi"
+		$nodeConfigValue | Add-Member -notepropertyname "productAttributes" -notepropertyvalue $esxiConfigTypeValues
+		$passwordPolicyNodes.Add($nodeConfigValue)
+		Remove-Variable -Name nodeConfigValue
+		Remove-Variable -Name esxiConfigTypeValues
+	}
+	return $passwordPolicyNodes
+}
+
+Function GetVCServerPasswordPolicy {
+	Param (
+		[Parameter (Mandatory = $true)] [ValidateNotNullOrEmpty()] [String]$server,
+        [Parameter (Mandatory = $true)] [ValidateNotNullOrEmpty()] [String]$user,
+        [Parameter (Mandatory = $true)] [ValidateNotNullOrEmpty()] [String]$pass,
+		[Parameter (Mandatory = $true)] [ValidateNotNullOrEmpty()] [String]$domain
+	)
+	
+	#construct Environment Object
+	$passwordPolicyNodes = New-Object System.Collections.Generic.List[System.Object]
+	$VCConfigTypeValues =  New-Object -TypeName psobject
+	$VCConfigTypeValues | Add-Member -notepropertyname "passwdExpInDays" -notepropertyvalue "Empty" 
+	$VCConfigTypeValues | Add-Member -notepropertyname "passwdNotifyEmail" -notepropertyvalue "Empty"
+	
+	$result = Get-VCServerPasswordPolicy -server $server -user $user -pass $pass -domain $domain
+	$VCConfigTypeValues.passwdNotifyEmail = $result.email
+	$VCConfigTypeValues.passwdExpInDays = $result.max_days_between_password_change
+	
+	$nodeConfigValue = New-Object -TypeName psobject
+	$nodeConfigValue | Add-Member -notepropertyname "FQDN" -notepropertyvalue $result.fqdn
+	$nodeConfigValue | Add-Member -notepropertyname "productType" -notepropertyvalue "VC"
+	$nodeConfigValue | Add-Member -notepropertyname "productAttributes" -notepropertyvalue $VCConfigTypeValues
+	$passwordPolicyNodes.Add($nodeConfigValue)
+	return $passwordPolicyNodes
+}
+
+Function GetSingleSignOnPasswordPolicy {
+	<#
+        Retrieves Single Sign On servers' Password Policies
+    #>
+	Param (
```

Comment:
Formatting, all Parameter values in line

Reply:
addressed in 9ee41934dd0ee5a302dd128339b72e2f82898175